### ISSUE TITLE
「BUG」Can‘t Exit OBS When Running Replay

### ISSUE THREAD

**Issue by gongyoyo** [obsproject/obs-studio#2162]
## Current Behavior
When start the replay butter,can't exit the progarm.

## Steps to Reproduce
1.Open OBS Studio 24.0.2
2.Start the replay butter button
3.Then minisize the window
4.Right clik the tray icon to select exit
5.Clik ''YES'' button in the next step

## Additional information
After these step.The progarm was be closed.
But the tray icon is still there.It's still working.
Also can't open the new OBS and can't right clik the tray icon.

**Comment by Fenrirthviti** [comment]
Can you please provide a log file from when this issue occurs? I am not able to replicate on my end.

**Comment by gongyoyo** [comment]
> Can you please provide a log file from when this issue occurs? I am not able to replicate on my end.

19:27:23.575: CPU Name: Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz
19:27:23.575: CPU Speed: 3600MHz
19:27:23.575: Physical Cores: 4, Logical Cores: 8
19:27:23.575: Physical Memory: 16342MB Total, 11748MB Free
19:27:23.575: Windows Version: 10.0 Build 17763 (revision: 805; 64-bit)
19:27:23.575: Running as administrator: true
19:27:23.575: Aero is Enabled (Aero is always on for windows 8 and above)
19:27:23.575: Windows 10 Gaming Features:
19:27:23.575: 	Game Bar: Off
19:27:23.576: 	Game DVR: Off
19:27:23.576: 	Game DVR Background Recording: Off
19:27:23.577: Sec. Software Status:
19:27:23.578: 
19:27:23.578: 
19:27:23.578: 
19:27:23.578: Current Date/Time: 2019-11-05, 19:27:23
19:27:23.578: Browser Hardware Acceleration: true
19:27:23.578: Portable mode: false
19:27:24.059: OBS 24.0.3 (64-bit, windows)
19:27:24.059: ---------------------------------
19:27:24.060: ---------------------------------
19:27:24.060: audio settings reset:
19:27:24.060: 	samples per sec: 48000
19:27:24.060: 	speakers:        2
19:27:24.061: ---------------------------------
19:27:24.061: Initializing D3D11...
19:27:24.061: Available Video Adapters: 
19:27:24.063: 	Adapter 0: NVIDIA GeForce GTX 1060 6GB
19:27:24.063: 	  Dedicated VRAM: 2051014656
19:27:24.063: 	  Shared VRAM:    4273209344
19:27:24.063: 	  output 0: pos={0, 0}, size={1920, 1080}, attached=true
19:27:24.065: Loading up D3D11 on adapter NVIDIA GeForce GTX 1060 6GB (0)
19:27:24.112: D3D11 loaded successfully, feature level used: b000
19:27:24.112: D3D11 GPU priority setup success
19:27:24.549: ---------------------------------
19:27:24.549: video settings reset:
19:27:24.549: 	base resolution:   1920x1080
19:27:24.549: 	output resolution: 1920x1080
19:27:24.549: 	downscale filter:  Lanczos
19:27:24.549: 	fps:               60/1
19:27:24.549: 	format:            NV12
19:27:24.549: 	YUV mode:          709/Partial
19:27:24.549: NV12 texture support enabled
19:27:24.550: Audio monitoring device:
19:27:24.550: 	name: XIBERIA MG-1 (Realtek High Definition Audio)
19:27:24.550: 	id: {0.0.0.00000000}.{d8a331bf-252a-4cc0-98c7-dcb692d166e4}
19:27:24.550: ---------------------------------
19:27:24.559: [CoreAudio encoder]: Adding CoreAudio AAC encoder
19:27:24.559: Failed to load 'en-US' text for module: 'decklink-ouput-ui.dll'
19:27:24.601: [AMF] AMF Test failed due to one or more errors.
19:27:24.601: Failed to initialize module 'enc-amf.dll'
19:27:24.619: [obs-browser]: Version 2.7.15
19:27:24.623: NVENC supported
19:27:24.678: Couldn't find VLC installation, VLC video source disabled
19:27:24.683: No blackmagic support
19:27:24.688: ---------------------------------
19:27:24.688:   Loaded Modules:
19:27:24.688:     win-wasapi.dll
19:27:24.688:     win-mf.dll
19:27:24.688:     win-dshow.dll
19:27:24.688:     win-decklink.dll
19:27:24.688:     win-capture.dll
19:27:24.688:     vlc-video.dll
19:27:24.688:     text-freetype2.dll
19:27:24.688:     rtmp-services.dll
19:27:24.688:     obs-x264.dll
19:27:24.688:     obs-vst.dll
19:27:24.688:     obs-transitions.dll
19:27:24.688:     obs-text.dll
19:27:24.688:     obs-qsv11.dll
19:27:24.688:     obs-outputs.dll
19:27:24.688:     obs-filters.dll
19:27:24.688:     obs-ffmpeg.dll
19:27:24.688:     obs-browser.dll
19:27:24.688:     image-source.dll
19:27:24.688:     frontend-tools.dll
19:27:24.688:     enc-amf.dll
19:27:24.688:     decklink-ouput-ui.dll
19:27:24.688:     coreaudio-encoder.dll
19:27:24.688: ---------------------------------
19:27:24.688: ==== Startup complete ===============================================
19:27:24.699: All scene data cleared
19:27:24.699: ------------------------------------------------
19:27:24.740: WASAPI: Device 'XIBERIA MG-1 (Realtek High Definition Audio)' initialized
19:27:24.772: adding 42 milliseconds of audio buffering, total audio buffering is now 42 milliseconds (source: XIBERIA MG-1)
19:27:24.772: 
19:27:24.788: WASAPI: Device 'NetEase Music (Realtek High Definition Audio)' initialized
19:27:24.813: WASAPI: Device 'Mix/Technica (Realtek High Definition Audio)' initialized
19:27:25.344: Switched to scene 'Battlefield 4'
19:27:25.345: ------------------------------------------------
19:27:25.345: Loaded scenes:
19:27:25.345: - scene 'NVIDIA GeForce® GTX':
19:27:25.345:     - source: '1#InPrivate' (image_source)
19:27:25.345:     - source: '1#NVIDIA' (monitor_capture)
19:27:25.345: - scene 'Counter Strike：Global Offensive':
19:27:25.345:     - source: '2#InPrivate' (image_source)
19:27:25.345:     - source: '2#NVIDIA' (game_capture)
19:27:25.345: - scene 'PLAYERUNKNOWN'S BATTLEGROUNDS':
19:27:25.345:     - source: '3#InPrivate' (image_source)
19:27:25.345:     - source: '3#NVIDIA' (game_capture)
19:27:25.345: - scene 'Minecraft':
19:27:25.345:     - source: '4#InPrivate' (image_source)
19:27:25.345:     - source: '4#NVIDIA' (game_capture)
19:27:25.345: - scene 'Battlefield 4':
19:27:25.345:     - source: '5#InPrivate' (image_source)
19:27:25.345:     - source: '5#NVIDIA' (game_capture)
19:27:25.345: - scene 'Tom Clancy's Rainbow Six Siege':
19:27:25.345:     - source: '6#InPrivate' (image_source)
19:27:25.345:     - source: '6#NVIDIA' (game_capture)
19:27:25.345: ------------------------------------------------
19:27:25.352: [game-capture: '5#NVIDIA'] attempting to hook process: bf4.exe
19:27:25.367: [game-capture: '5#NVIDIA'] shared texture capture successful
19:27:30.148: [jim-nvenc: 'recording_h264'] settings:
19:27:30.148: 	rate_control: VBR
19:27:30.148: 	bitrate:      10240
19:27:30.148: 	cqp:          20
19:27:30.148: 	keyint:       120
19:27:30.148: 	preset:       mq
19:27:30.148: 	profile:      high
19:27:30.148: 	width:        1920
19:27:30.148: 	height:       1080
19:27:30.148: 	2-pass:       true
19:27:30.148: 	b-frames:     2
19:27:30.148: 	lookahead:    true
19:27:30.148: 	psycho_aq:    true
19:27:30.148: 
19:27:30.170: [CoreAudio AAC: 'Track2']: settings:
19:27:30.170: 	mode:          AAC
19:27:30.170: 	bitrate:       320
19:27:30.170: 	sample rate:   48000
19:27:30.170: 	cbr:           on
19:27:30.170: 	output buffer: 1536
19:27:30.173: ==== Replay Buffer Start ===========================================
19:27:39.739: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:27:39.753: [game-capture: '5#NVIDIA'] shared texture capture successful
19:27:39.758: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:27:39.762: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:27:39.767: [game-capture: '5#NVIDIA'] shared texture capture successful
19:29:02.751: [game-capture: '5#NVIDIA'] Hooked D3D9
19:31:17.841: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:17.860: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:17.867: [game-capture: '5#NVIDIA'] shared texture capture successful
19:31:18.689: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:18.704: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:18.717: [game-capture: '5#NVIDIA'] shared texture capture successful
19:31:18.722: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:18.727: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:18.734: [game-capture: '5#NVIDIA'] shared texture capture successful
19:31:18.977: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:18.984: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:19.000: [game-capture: '5#NVIDIA'] shared texture capture successful
19:31:26.329: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:26.341: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:26.350: [game-capture: '5#NVIDIA'] shared texture capture successful
19:31:26.371: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:26.376: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:26.384: [game-capture: '5#NVIDIA'] shared texture capture successful
19:31:29.139: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:29.142: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:29.160: [game-capture: '5#NVIDIA'] shared texture capture successful
19:31:33.349: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:33.366: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:33.367: [game-capture: '5#NVIDIA'] shared texture capture successful
19:31:33.396: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:33.402: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:33.417: [game-capture: '5#NVIDIA'] shared texture capture successful
19:31:44.308: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:44.327: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:44.334: [game-capture: '5#NVIDIA'] shared texture capture successful
19:31:44.994: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:45.011: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:45.017: [game-capture: '5#NVIDIA'] shared texture capture successful
19:31:45.035: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:31:45.038: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:31:45.050: [game-capture: '5#NVIDIA'] shared texture capture successful
19:33:01.529: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:33:01.550: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:33:01.550: [game-capture: '5#NVIDIA'] shared texture capture successful
19:33:03.310: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:33:03.323: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:33:03.334: [game-capture: '5#NVIDIA'] shared texture capture successful
19:33:03.346: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:33:03.351: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:33:03.367: [game-capture: '5#NVIDIA'] shared texture capture successful
19:33:44.461: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:33:44.481: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:33:44.484: [game-capture: '5#NVIDIA'] shared texture capture successful
19:33:44.996: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:33:45.007: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:33:45.017: [game-capture: '5#NVIDIA'] shared texture capture successful
19:33:45.031: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:33:45.035: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:33:45.050: [game-capture: '5#NVIDIA'] shared texture capture successful
19:33:45.779: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:33:45.798: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:33:45.800: [game-capture: '5#NVIDIA'] shared texture capture successful
19:33:52.048: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:33:52.064: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:33:52.067: [game-capture: '5#NVIDIA'] shared texture capture successful
19:33:52.096: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:33:52.101: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:33:52.117: [game-capture: '5#NVIDIA'] shared texture capture successful
19:34:54.796: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:34:54.817: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:34:54.817: [game-capture: '5#NVIDIA'] shared texture capture successful
19:34:55.948: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:34:55.963: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:34:55.967: [game-capture: '5#NVIDIA'] shared texture capture successful
19:34:55.986: [game-capture: '5#NVIDIA'] ----------------- d3d11 capture freed ----------------
19:34:55.991: [game-capture: '5#NVIDIA'] d3d11 shared texture capture successful
19:34:56.000: [game-capture: '5#NVIDIA'] shared texture capture successful
19:37:48.283: [game-capture: '5#NVIDIA'] capture window no longer exists, terminating capture
19:37:48.284: [game-capture: '5#NVIDIA'] capture stopped
19:37:53.108: ==== Shutting down ==================================================
19:37:53.117: Switched to scene '(null)'
19:37:53.123: WASAPI: Device 'XIBERIA MG-1 (Realtek High Definition Audio)' Terminated
19:37:53.130: WASAPI: Device 'NetEase Music (Realtek High Definition Audio)' Terminated
19:37:53.136: WASAPI: Device 'Mix/Technica (Realtek High Definition Audio)' Terminated
19:37:53.141: All scene data cleared
19:37:53.141: ------------------------------------------------
19:37:53.151: Output '回放缓存': stopping
19:37:53.151: Output '回放缓存': Total frames output: 37361
19:37:53.151: Output '回放缓存': Total drawn frames: 37378 (37379 attempted)
19:37:53.151: Output '回放缓存': Number of lagged frames due to rendering lag/stalls: 1 (0.0%)
19:37:53.152: Video stopped, number of skipped frames due to encoding lag: 1/37377 (0.0%)
![批注_2019-11-05_19-38-32](https://user-images.githubusercontent.com/36725275/68204951-16c37980-0004-11ea-9ddf-11d165094c34.png)

**Comment by Fenrirthviti** [comment]
Then you try to exit OBS with an output active, it prompts you to exit:

https://pub.rachni.com/img/obs64_2019-11-05_16-21-12.png

This dialog is likely just opening behind another window. No bug.

In the future, please submit support requests through the appropriate channels.

**Comment by Fenrirthviti** [comment]
On further testing, I was able to replicate this, but I'm not sure the exact steps to cause it to happen. It seems to be somewhat random.

**Comment by xcasxcursex** [comment]
I'm also experiencing this, and yes it does appear to happen at random.

Is there a way I can help here? Is there a method to have OBS provide more verbose logs? Since mine fails about half the time, it wouldn't take me long to provide logs which pin it down.

**Comment by xcasxcursex** [comment]
More info on this: TL;DR I can't disable the systray icon. It's not enabled, but there it is.

Noticing that OP mentioned that a necessary step was to minimise to the tray, I decided to disable the systray for the time being. (also noticed this was reported back in 24.0.2, I'm on 24.0.3.
I started the app, unchecked the box, clicked OK, the systray icon disappeared, and exited as normal.
Started again to test, and the systray icon is there. Go into settings, box is unticked.
I have to tick to enable it, hit OK (not 'apply'! That's not good enough for it to work) and then go into settings and untick the box and hit OK (not apply), every time I launch, to make the systray icon go away.
I've confirmed that my changes are making it into the settings file, even when I only hit apply (which does not apply those settings)
I figured it has to be something wrong with my settings so I delete them all (backed up first of course).... and it works like this out of the box. Start with an empty obs-studio directory, say no to autoconfig, disable systray....As soon as I start recording, systray icon appears and it doesn't go away when I stop recording (although the red dot disappears)

Not sure if this is a separate bug but it strikes me that it might just fall under the same umbrella.

**Comment by kkartaltepe** [comment]
Small follow up:

After hitting apply the taskbar tray icon choice will take place on either OK or Cancel, this seems working as intended (after Apply the pending changes cannot be Cancel'ed). I agree it probably should take affect on Apply and im not sure why it waits until the dialog is closed to update.

**Comment by CodyJung** [comment]
I caught this (or a similar issue?) on the debugger but am basically brand new to the code, so I'm not sure what the right fix is.

The main thread is waiting on Replay Buffer's `stopping_event` to be signaled. In `obs_output_actual_stop`, `force` is true, and `delay_active` is false, so it falls into the `call_stop` case which calls out to `ffmpeg_mux_stop()`. For some reason it never gets the stopping_event signal, so just hangs out.

My (100%) reproduction steps on Windows 64 are as follows:
1. Choose Start Recording
2. Choose Start Replay Buffer
3. Click the Pause icon next to stop recording.
4. Close the popup and click Exit, OR right-click the tray icon, choose Exit
5. Click Yes

**Comment by xcasxcursex** [comment]
Experiencing this after updating to 28 and popped by to report the regression, but perhaps it's just still there from before....

**Comment by xcasxcursex** [comment]
Steps to reproduce:

- Download fresh OBS from https://github.com/obsproject/obs-studio/releases/download/28.1.2/OBS-Studio-28.1.2-Full-x64.zip
- Create a shortcut and add --portable --startreplaybuffer
- Start it from said shortcut, and run the wizard preferring recording and (I doubt this part is important, but for completeness, this was my process) using the largest desktop's size as canvas size and 60FPS as framerate.
- Complete the wizard, open settings, Enable: Replay buffer, minimize to tray when started, Disable: warn when active outputs
- Exit OBS and then start it up using the shortcut above
- Right-click the tray icon and select Exit.
- Hang. Process is still running, tray icon is visible, but the application is unresponsive.

Logs of these two startups (probably only the second one, where it crashes, is of interest, but both for completeness)
Setup:
[2022-11-20 18-19-51.txt](https://github.com/obsproject/obs-studio/files/10049073/2022-11-20.18-19-51.txt)
Start and exit/hang:
[2022-11-20 18-20-59.txt](https://github.com/obsproject/obs-studio/files/10049072/2022-11-20.18-20-59.txt)

Since the analyzer suggests them both, I should add that HAGS and admin have no effect here.

**Comment by Fenrirthviti** [comment]
Closing as resolved as this hasn't had updates in many years, and we have made significant changes to the affected areas. If this is still a problem, please open a new issue report on the current version with updated reproduction steps.
